packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2704,14 +2710,75 @@ class _Location {
   }
 }
 
+/// Describer to gather important information about [debugCreator].
+///
+/// This function will be attached to [FlutterErrorDetails.describeCreator]
+/// in [WidgetsBinding].
+Iterable<DiagnosticsNode> describeCreator(dynamic debugCreator) {
+  final Element element = debugCreator is Element? debugCreator: debugCreator.element;
+  return <DiagnosticsNode>[
+    _describeRelevantUserCode(element),
+  ];
+}
+
+DiagnosticsNode _describeRelevantUserCode(Element element) {
+  const String name = 'User created ancestor of the error causing widget was';
+  if (!WidgetInspectorService.instance.isWidgetCreationTracked()) {
+    return StringProperty(
+      name,
+      '\nWidget creation tracking is currently disabled. Enabling it would lead to improved error messages.'
+      ' It can be enabled by passing `--track-widget-creation` to `flutter run` or `flutter test`',
+      quoted: false,
+    );
+  }
+  DebugElementWrapper node = DebugElementWrapper(element);
```

Comment:
This DebugElementWrapper class seems odd.
You should use
element.visitAncestorElements
or similar existing API.